sdk/core/amqp-common/src/retry.ts

```diff
@@ -97,12 +97,22 @@ async function checkNetworkConnection(host: string): Promise<boolean> {
   if (isNode) {
     return new Promise((res) => {
       resolve(host, function(err: any): void {
-        // List of possible DNS error codes: https://nodejs.org/dist/latest-v12.x/docs/api/dns.html#dns_error_codes
-        if (err && (err.code === "ECONNREFUSED" || err.code === "ETIMEOUT")) {
-          res(false);
+        if (err) {
+          log.retry(
+            "Error thrown from dns.resolve in network connection check: '%s', %O",
+            err.code || err.name,
+            err
+          );
+
+          // List of possible DNS error codes: https://nodejs.org/dist/latest-v12.x/docs/api/dns.html#dns_error_codes
+          if (err.code === "ECONNREFUSED" || err.code === "ETIMEOUT") {
```

Comment:
Are these constants available directly from the 'dns' module? Or do we have to use the string constants?

Reply:
They are, but trying to use them makes rollup complain. When I make the same change to core-amqp I'll do some refactoring to make this more tidy.